src/lib/features/group/createGroupService.ts

```diff
@@ -1,21 +1,28 @@
 import { IUnleashConfig } from '../../types';
-import { GroupService } from '../../services';
+import { EventService, GroupService } from '../../services';
 import { Db } from '../../db/db';
 import GroupStore from '../../db/group-store';
 import { AccountStore } from '../../db/account-store';
 import EventStore from '../../db/event-store';
+import FeatureTagStore from '../../db/feature-tag-store';
 
 export const createGroupService = (
     db: Db,
     config: IUnleashConfig,
 ): GroupService => {
-    const { getLogger } = config;
+    const { getLogger, eventBus } = config;
     const groupStore = new GroupStore(db);
     const accountStore = new AccountStore(db, getLogger);
     const eventStore = new EventStore(db, getLogger);
+    const featureTagStore = new FeatureTagStore(db, eventBus, getLogger);
+    const eventService = new EventService(
+        { eventStore, featureTagStore },
+        { getLogger },
+    );
```

Comment:
I disagree with this one due to the fact that other `createXService` methods follow the same approach: They create the needed services inside the methods instead of accepting them as arguments.

If we decide to refactor this I would prefer to target all of those methods and do it in a separate PR.

Reply:
I can explain the details behind createXService methods. Those are "composition roots" for each service. So they hide the information how each service is created. Some of our services depend on other services (which is often the consequence of not have read models). If creating a dependent service is difficult I usually create another composition root (createXService) method for the dependent service and call it from my composition root. On the other hand some services are trivial to instantiate or I didn't have time to migrate them to compositon roots. So my recommendation is to decide on a case by case basis. If you feel the pain of constructing a complex graph of objects: create a compositon root for a dependent service.

Reply:
> they hide the information how each service is created

I think this implies that they should not know how other services are created other than themselves. It acts as a Dependency Injection layer, where services get services injected rather than creating them on each composition root, which might lead to maintainability issues when having to modify how a service is created...

> On the other hand some services are trivial to instantiate or I didn't have time to migrate them to composition roots. 

I believe this one is simple, but I'm ok with whatever Nuno decides to do here, we'll have a chat and move this forward.

Reply:
Also, in other places we prefer injecting the service: https://github.com/Unleash/unleash/pull/4830/files#diff-0f55a0a1460b7ad5fd21da38762c259c1580e877e4b6b746e0032ad27776c886R93 but as Mateusz said, it can be considered tech debt for later